lib/WebService/GrowthBook.pm

```diff
@@ -0,0 +1,160 @@
+package WebService::GrowthBook;
+# ABSTRACT: ...
+
+use strict;
+use warnings;
+no indirect;
+use feature qw(state);
+use Object::Pad;
+use JSON::MaybeUTF8 qw(decode_json_text);
+use Scalar::Util qw(blessed);
+use WebService::GrowthBook::FeatureRepository;
+use WebService::GrowthBook::Feature;
+use WebService::GrowthBook::FeatureResult;
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::GrowthBook - sdk of growthbook
+
+=head1 SYNOPSIS
+
+    use WebService::GrowthBook;
+    my $instance = WebService::GrowthBook->new(client_key => 'my key');
+    $instance->load_features;
+    if($instance->is_on('feature_name')){
+        # do something
+    }
+    else {
+        # do something else
+    }
+    my $string_feature = $instance->get_feature_value('string_feature');
+    my $number_feature = $instance->get_feature_value('number_feature');
+    # get decoded json
+    my $json_feature = $instance->get_feature_value('json_feature');
+
+=head1 DESCRIPTION
+
+    This module is a sdk of growthbook, it provides a simple way to use growthbook features.
+
+=cut
+
+# singletons
+my  $feature_repository = WebService::GrowthBook::FeatureRepository->new;
+
+class WebService::GrowthBook {
+    field $url: param //= 'https://api.growthbook.io/api/v1';
+    field $client_key: param //= '';
+    field $features: param //= {};
+
+    method load_features {
+        if(!$client_key) {
+            die "Must specify 'client_key' to refresh features";
+        }
+        my $loaded_features = $feature_repository->load_features($url, $client_key);
+        if($loaded_features){
+            $self->set_features($loaded_features);
+        }
+        return 1;
+    }
+    method set_features($features_set) {
+        $features = {};
+        for my $feature ($features_set->@*) {
+            if(blessed($feature) && $feature->isa('WebService::GrowthBook::Feature')){
+                $features->{$feature->id} = $feature;
+            }
+            else {
+                $features->{$feature->{id}} = WebService::GrowthBook::Feature->new(id => $feature->{id}, default_value => $feature->{defaultValue}, value_type => $feature->{valueType});
+            }
+        }
+    }
+    
+    method is_on($feature_name) {
+        # TODO how to do if no such feature name?
```

Comment:
We should return False if feature name is not available. We need to follow white listed approach.

Reply:
return false is not suitable since it cannot differ  from the value false.
maybe undef will be better.
Now an exception will be thrown when do `eval_feature`. we can remove this  TODO line now.